### ISSUE TITLE
Fallback for missing BCD data

### ISSUE THREAD

**Issue by jpmedley** [GoogleChrome/web.dev#6039]
**Is your feature request related to a problem? Please describe.**
Sometimes it's desirable to use the new BCD widget in before the data is available on MDN. (See [this PR for an example](https://github.com/GoogleChrome/web.dev/pull/6033/files#diff-d3d47ffb11ca3171b67d84756e8bc7a144c581ca9c2f50f8f5a1ce77914c55e7R49).)

**Describe the solution you'd like**
It would be useful if we could pass a fallback value to the widget for missing data. In the example PR above, we know the feature is shipping in Chrome 94 because the intent has it's [three LGTMs](https://groups.google.com/a/chromium.org/g/blink-dev/c/ZmHHfrR_jak/m/YXHH1ls1AAAJ). We just don't know when this information will be available in MDN's BCD database. 

**Describe alternatives you've considered**
A few options:
* Try to remember weeks from now that we need to go back update a web.dev article.
* We could also pass the widget something more vague than a number, something like an argument that will make it render "Complete" until the data is available.

**Additional context**
Long-term, I'm worried about the accumulation of places where Fugu-style progress tables should be replaced but aren't. A feature like this would allow authors to update articles at what is for them the most convenient time. The linked PR is an example of this.

**Comment by jpmedley** [comment]
cc: @tomayac

**Comment by tomayac** [comment]
Thanks for looping me in, @jpmedley! @devnook, I think a fallback that expresses somehow that compatibility data is pending would make the most sense. Like that the widget would just magically update once the data becomes available.

**Comment by jpmedley** [comment]
"Pending" would imply that we're working on it. The fact is that "Pending" could be up months if no one can find time to prioritize BCD. That could also be misread as "it will be in Chrome, but not yet."

I'm trying to address a very narrow issue: we know a feature will be in a specific version because it has the required approvals.

**Comment by tomayac** [comment]
If "pending" raises hopes to high, maybe just displaying "n/a" in the widget could work.

**Comment by jpmedley** [comment]
And what does that mean?

**Comment by tomayac** [comment]
That no browser compatibility data is available (yet). 
¯\_(ツ)_/¯

**Comment by samthor** [comment]
To chime in as TL, I think something like "Not available" makes sense here, perhaps with a local build-time warning that something is missing (so if we end up with 100s it's interesting, rather than a harsher build failure).

It's also possible that BCD data _goes away_, say for a very old API.

**Comment by devnook** [comment]
Re: "You know that BCD data will be available at api.FooInterface."

Is it usual that we know exactly *where* the FooInterface will end up? e.g. Tat it will not go into api.someSubsection.FooInterface?

I'm sorry if I ask an uninformed question, but this is key to decide if we allow such widget to render or we discard it as incorrect.

**Comment by tomayac** [comment]
My understanding is that the names are directly based on the IDL from the spec, for example, [`PeriodicSyncManager`](https://wicg.github.io/periodic-background-sync/#periodicsyncmanager-interface), which is then used for the [compatibility table](https://developer.mozilla.org/en-US/docs/Web/API/Web_Periodic_Background_Synchronization_API#browser_compatibility).

**Comment by jpmedley** [comment]
This pattern is always api._InterfaceName_.[_memberName_]. It seems unlikely that the keys would ever move since this matches the structure of APIs, the structure of IDL files, and the site structure of MDN itself.

**Comment by jpmedley** [comment]
These keys appear on every API reference page. For MDN to ever change this would be enormously painful.

**Comment by jpmedley** [comment]
@samthor Think about what a page might potentially look like if you say "Not Available". You could have text in the article itself, maybe next to the macro, mentioning a version number. Readers will just scratch their heads and say, "What?". I don't think we can absolutely exclude such references from the text. As soon as we say that, we'll start running into cases where it makes no sense.

**Comment by rachelandrew** [comment]
Given that we are moving all experimental content to DCC, I think that there should not be new instances on web.dev where the widget has no data. So it may not be worth working on a fallback.

**Comment by jpmedley** [comment]
Isn't @devnook working on feature parity between the two sites? Wouldn't that mean common code between the two sites? Are you saying we would never use this widget on an in-development feature?

**Comment by jpmedley** [comment]
Do I need to transfer this ticket to dcc?

**Comment by rachelandrew** [comment]
I don't think it's useful for an experimental feature as you already outlined. On DCC they are more likely to have the kind of status table we use (as things progress through origin trial etc.) so I think it's likely a case of this is a web.dev feature, for things with stable browser support.

**Comment by jpmedley** [comment]
I disagree. As new features progress actual data will gradually override the fallback data. We won't need to do anything.

**Comment by jpmedley** [comment]
I opened this ticket specifically because this macro has utility in cutting edge features.

**Comment by tomayac** [comment]
The bot is about to close this. Have we come to a conclusion? It sounds like migrating the ticket to the DCC is the short-term thing to do.

**Comment by rachelandrew** [comment]
Commenting removes the stale flag, I think it's fine to leave here for now. Whatever we do with the widget is likely to apply to both sites anyway as we're trying to make all the components the same.
